### PR TITLE
add roster Excel parser with tests and template

### PR DESCRIPTION
**Pull request by Lillian2016**
1. Introduce a roster Excel parser with structured issues and header alias support.
2. Add unit tests for parsing, warnings, and time/date handling.
3. Include a roster template and update shared docs/deps.

## Summary by CodeRabbit

* **New Features**
  * Excel-based roster import: shift & employee parsing, validation, computed summaries, and STRICT/LENIENT modes.
  * Header normalization, duplicate detection, issue grouping with samples for easier remediation.

* **Documentation**
  * Comprehensive roster import guide with format specs, templates, examples, and troubleshooting.

* **Tests**
  * Extensive unit/integration tests covering parsing, validation, utilities, and edge cases.

* **Chores**
  * Pydantic configured with email extras; Excel support added via openpyxl.
  * Consolidated/ deprecated legacy shared exports in favor of a unified shared surface.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by Zephyr591 — DISMISSED** [review]

File: agent-service/agents/roster/services/roster_import/row_parsers.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+        )
+
+    entry = EmployeeEntry(
+        excel_row=row_num,
+        name=name,
+        email=email,
+        role=role,
+        department=get_string(normalized.get("department")),
+        start_date=parse_date(normalized.get("start_date")),
+    )
```
[the remark this answers is not in the document]
Comment (Lillian2016) [reply agent-service/agents/roster/services/roster_import/row_parsers.py]: Wrapped start_date parsing in [row_parsers.py] with a try/except and return a proper INVALID_DATE ParseIssue (row/column/value included).

File: backend/scripts/init-migrations-and-database.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Write-Host "Migrations folder deleted" -ForegroundColor Gray
+    }
+
+    # Step 3: Create fresh InitialCreate migration
+    Write-Host "Creating InitialCreate migration..." -ForegroundColor Yellow
+    dotnet ef migrations add InitialCreate --project src/FairWorkly.Infrastructure --startup-project src/FairWorkly.API
+
+    # Step 4: Apply migration to create new database
+    Write-Host "Applying migration..." -ForegroundColor Yellow
+    dotnet ef database update --project src/FairWorkly.Infrastructure --startup-project src/FairWorkly.API
```
[the remark this answers is not in the document]
Comment (Lillian2016) [reply backend/scripts/init-migrations-and-database.ps1]: It does not belong to the code submitted this time.

File: backend/scripts/reset-database.ps1 [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+Push-Location $BackendDir
+
+try {
+    Write-Host "Dropping database..." -ForegroundColor Yellow
+    dotnet ef database drop --force --project src/FairWorkly.Infrastructure --startup-project src/FairWorkly.API
+
+    Write-Host "Applying migrations..." -ForegroundColor Yellow
+    dotnet ef database update --project src/FairWorkly.Infrastructure --startup-project src/FairWorkly.API
+
+    Write-Host "Database reset complete!" -ForegroundColor Green
```
[the remark this answers is not in the document]
Comment (Lillian2016) [reply backend/scripts/reset-database.ps1]: It does not belong to the code submitted this time.

File: backend/scripts/update-database.ps1 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+$BackendDir = Split-Path -Parent $ScriptDir
+
+Write-Host "Working directory: $BackendDir" -ForegroundColor Cyan
+Push-Location $BackendDir
+
+try {
+    Write-Host "Applying pending migrations..." -ForegroundColor Yellow
+    dotnet ef database update --project src/FairWorkly.Infrastructure --startup-project src/FairWorkly.API
+
+    Write-Host "Database update complete!" -ForegroundColor Green
```
[the remark this answers is not in the document]
Comment (Lillian2016) [reply backend/scripts/update-database.ps1]: It does not belong to the code submitted this time.

File: backend/src/FairWorkly.API/Program.cs [hunk 5]
```diff
@@ -123,27 +133,28 @@
...
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = key,
+                ValidateIssuer = !string.IsNullOrEmpty(jwtIssuer),
+                ValidIssuer = jwtIssuer,
+                ValidateAudience = !string.IsNullOrEmpty(jwtAudience),
+                ValidAudience = jwtAudience,
+                ValidateLifetime = true,
+                ClockSkew = TimeSpan.FromSeconds(30),
+            };
+        });
```
[the remark this answers is not in the document]
Comment (Lillian2016) [reply backend/src/FairWorkly.API/Program.cs]: It does not belong to the code submitted this time.

**Review by Jason26214 — APPROVED** [review]
lgtm
